src/OpenTelemetry.Exporter.OpenTelemetryProtocol/OtlpTraceExporterHelperExtensions.cs

```diff
@@ -43,19 +44,31 @@ public static TracerProviderBuilder AddOtlpExporter(this TracerProviderBuilder b
         /// <param name="configure">Callback action for configuring <see cref="OtlpExporterOptions"/>.</param>
         /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
         public static TracerProviderBuilder AddOtlpExporter(this TracerProviderBuilder builder, Action<OtlpExporterOptions> configure)
-            => AddOtlpExporter(builder, name: null, configure);
+            => AddOtlpExporter(builder, name: null, configure, persistentStorageFactory: null);
+
+        /// <summary>
+        /// Adds OpenTelemetry Protocol (OTLP) exporter to the TracerProvider.
+        /// </summary>
+        /// <param name="builder"><see cref="TracerProviderBuilder"/> builder to use.</param>
+        /// <param name="configure">Callback action for configuring <see cref="OtlpExporterOptions"/>.</param>
+        /// <param name="persistentStorageFactory">Factory for creating an offline storage implementation.</param>
+        /// <returns>The instance of <see cref="TracerProviderBuilder"/> to chain the calls.</returns>
+        public static TracerProviderBuilder AddOtlpExporter(this TracerProviderBuilder builder, Action<OtlpExporterOptions> configure, Func<IServiceProvider, PersistentBlobProvider> persistentStorageFactory)
+            => AddOtlpExporter(builder, name: null, configure, persistentStorageFactory);
```

Comment:
Does it make sense to ship these extensions in a separate package - at least for now? e.g., `OpenTelemetry.Exporter.OpenTelemetryProtocol.PersistentStorage`

Would probably require using `InternalsVisibleTo`, but would enable shipping this functionality as experimental and not block stable releases of the OTLP exporter.

Reply:
Yup. separating would be the best approach now to keep things shipped as stable.

Reply:
Cool, made a new project :)